### PR TITLE
[jit] Fix builtin function reference

### PR DESCRIPTION
**Pull request by driazati**
This was previously buggy and not being displayed on master. This fixes
the issues with the script to generate the builtin function schemas and
moves it to its own page (it's 6000+ lines of schemas)

Sphinx looks like it will just keep going if it hits errors when importing modules, we should find out how to turn that off and put it in the CI

This also includes some other small fixes:
* removing internal only args from `script()` and `trace()` docs, this also requires manually keeping these argument lists up to date but I think the cleanliness is worth it
* removes outdated note about early returns

Differential Revision: [D16742406](https://our.internmc.facebook.com/intern/diff/16742406/)

### CODE REVIEW

File: torch/jit/supported_ops.py [hunk 1]
```diff
@@ -40,6 +41,47 @@ def hidden(name):
 
     functions = []
 
+    # Iterate over torch.nn.functional
+    mod = torch.nn.functional
+    name = mod.__name__
+    for elem in dir(torch.nn.functional):
+        # weak script functions
```
Comment (eellison) [inline torch/jit/supported_ops.py]: `weak script functions` ?

File: torch/jit/supported_ops.py [hunk 2]
```diff
@@ -40,6 +41,47 @@ def hidden(name):
 
     functions = []
 
+    # Iterate over torch.nn.functional
+    mod = torch.nn.functional
```
Comment (eellison) [inline torch/jit/supported_ops.py]: It's hard for me to review this without seeing its output. Anyway you could post it?

File: torch/jit/__init__.py [hunk 3]
```diff
@@ -1988,14 +1990,81 @@ def forward(self, input):
...
+
 _builtin_table = None
 
 _modules_containing_builtins = (torch, torch._C._nn)
 
-
-def _unwrap_optional(x):
-    assert x is not None, "Unwrapping null optional"
-    return x
+_builtin_ops = [
```
Comment (eellison) [inline torch/jit/__init__.py]: why was this changed in the PR?

File: torch/jit/__init__.py [hunk 4]
```diff
@@ -1988,14 +1990,81 @@ def forward(self, input):
...
+    (torch.nn.functional.interpolate, "aten::__interpolate"),
+    (torch.nn.functional.upsample_bilinear, "aten::__upsample_bilinear"),
+    (torch.nn.functional.upsample_nearest, "aten::__upsample_nearest"),
+    (torch.nn.functional.upsample, "aten::__upsample"),
+    (torch.nn.init._no_grad_fill_, "aten::_no_grad_fill_"),
+    (torch.nn.init._no_grad_normal_, "aten::_no_grad_normal_"),
+    (torch.nn.init._no_grad_uniform_, "aten::_no_grad_uniform_"),
+    (torch.nn.init._no_grad_zero_, "aten::_no_grad_zero_"),
+    (torch._C._get_tracing_state, "aten::_get_tracing_state"),
+    (warnings.warn, "aten::warn"),
```
Comment (eellison) [inline torch/jit/__init__.py]: This won't be a full set of built in ops, there are also all of these https://github.com/pytorch/pytorch/blob/4d676d53a67e610689afbb34deabcac6ba8b8c50/torch/csrc/jit/script/compiler.cpp#L390

File: torch/jit/__init__.py (hunk 3 above)
Comment (driazati) [reply torch/jit/__init__.py]: to be accessible from `torch/jit/supported_ops.py`

File: torch/jit/__init__.py (hunk 4 above)
Comment (driazati) [reply torch/jit/__init__.py]: Right now the builtins list throws an error when running so it isn't included in the docs at all, I think we can leave these for a follow up

**Review by eellison — APPROVED** [review]
Builtin functions look good to me, I don't know that it makes sense to include the `torch.nn.functional` functions since we support all of them. If you look at the `Builtin Functions` section [here](http://172.24.100.69:8086/jit.html#migrating-to-pytorch-1-2-recursive-scripting-api) we only list unsupported functions. Maybe we could list this page in that section as well. 

Other notes (follow ups): 
it would be nice to break out the builtin python modules, (and potentially types as well).

Separate sections for 
`math`
string methods,
list methods, 
dict methods.
